sstables/index_reader.hh

```diff
@@ -441,6 +444,13 @@ class index_reader {
                 _sstable->_index_file_size);
     }
 
+    seastar::shared_ptr<cached_file> get_local_cached_file() {
+        if (!_local_cached_file) {
+            _local_cached_file = make_cached_file(make_tracked_index_file(_sstable->uncached_index_file()));
+        }
+        return _local_cached_file;
+    }
+
     seastar::shared_ptr<cached_file> get_cached_file() {
```

Comment:
I think get_cached_file() should use _local_cached_file if it's available, so that we don't have two separate caches for the partition index and bsearch cursor.

Reply:
Yes, good catch. It was supposed to, I must have lost the hunk during cleanup/rebase.